### DISCUSSION TITLE
Why GetReports return 0?

### DISCUSSION THREAD

**Discussion by hopeko** [abuzuhri/Amazon-SP-API-CSharp#56] [Q&A] [answered]
I want to get transaction data(Reports->Payments->Transaction View).

but reports.Count = 0
Why reports.Count = 0 ?

```cs
var parameters = new ParameterReportList();
parameters.pageSize = 100;
parameters.reportTypes = new List<ReportTypes>();
parameters.reportTypes.Add(ReportTypes.GET_VAT_TRANSACTION_DATA);
parameters.marketplaceIds = new List<string>();
parameters.marketplaceIds.Add(MarketPlace.US.ID);
var reports = amazonConnection.Reports.GetReports(parameters);
```

**Comment by abuzuhri** [comment]
This report available only to some country not to all countries

**Reply by hopeko** [reply]
> This report available only to some country not to all countries

I have tried another method, 

```cs
var parameters = new ParameterListFinancialEvents();
parameters.PostedAfter = new DateTime(2021, 12, 1);
var fEvents = amazonConnection.Financial.ListFinancialEvents(parameters);
```

I can get CouponPaymentEventList, CouponPaymentEventList, ProductAdsPaymentEventList, RefundEventList, ServiceFeeEventList, ShipmentEventList, but the count of all the other EventList is 0, for example SellerDealPaymentEventList. Why is that?

**Reply by abuzuhri** [reply]
Most probably it's as it's came from Amazon but I will double check from my side to confirm

**Reply by abuzuhri** [reply]
> ```cs
> var parameters = new ParameterListFinancialEvents();
> parameters.PostedAfter = new DateTime(2021, 12, 1);
> var fEvents = amazonConnection.Financial.ListFinancialEvents(parameters);
> ```

Yes , The data here as its came from Amazon

**Reply by hopeko** [reply]
> > ```cs
> > var parameters = new ParameterListFinancialEvents();
> > parameters.PostedAfter = new DateTime(2021, 12, 1);
> > var fEvents = amazonConnection.Financial.ListFinancialEvents(parameters);
> > ```
> 
> Yes , The data here as its came from Amazon

Why can't I get data if it came from Amazon?
Isn't all data the API return is from Amazon?

**Reply by abuzuhri** [reply]
I mean data here as its came from Amazon

**Comment by abuzuhri** [comment]
I Upload new sample test here without use any API just pure c# 

https://github.com/abuzuhri/Amazon-SP-API-CSharp/tree/main/Others/PureCodeSampleForTest

please test the result you get with the result came from my Library and let me know

**Reply by hopeko** [reply]
Thanks! I will try later.

Now, I got the payment data I want.
```cs
var parameters = new ParameterReportList();
parameters.createdSince = new DateTime(2021, 12, 25);
parameters.pageSize = 100;
parameters.reportTypes = new List<ReportTypes>();
parameters.reportTypes.Add(ReportTypes.GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2);
parameters.marketplaceIds = new List<string>();
parameters.marketplaceIds.Add(MarketPlace.US.ID);
var reports = amazonConnection.Reports.GetReports(parameters);

foreach (var report in reports)
{
    var rep = amazonConnection.Reports.GetReportFile(report.ReportDocumentId);
    Console.WriteLine(rep);
}
```
But there is still a problem. I set `parameters.createdSince = new DateTime(2021, 12, 25);` but I got data from 12, 22, 2021 to 1, 4, 2022, Why I got data before 12, 25, 2021 and no data after 1, 4, 2022 in the meantime.(it's supposed to get data from 12, 25, 2021 to 1, 6, 2022)

**Reply by abuzuhri** [reply]
> ```cs
> var parameters = new ParameterReportList();
> parameters.createdSince = new DateTime(2021, 12, 25);
> parameters.pageSize = 100;
> parameters.reportTypes = new List<ReportTypes>();
> parameters.reportTypes.Add(ReportTypes.GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2);
> parameters.marketplaceIds = new List<string>();
> parameters.marketplaceIds.Add(MarketPlace.US.ID);
> var reports = amazonConnection.Reports.GetReports(parameters);
> 
> foreach (var report in reports)
> {
>     var rep = amazonConnection.Reports.GetReportFile(report.ReportDocumentId);
>     Console.WriteLine(rep);
> }
> ```



Also check this method as replacement for report generation

```CSharp
        amazonConnection.Reports.DownloadExistingReportAndDownloadFile(ReportTypes.GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2, fromDate, toDate);
```

**Comment by abuzuhri** [comment, marked as answer]
Depend on my experience on this report are the same report in this page GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2

https://sellercentral.amazon.ae/payments/past-settlements?ref_=xx_settle_ttab_dash

so you will not get statement on the date range you will get list of report in that date range

**Reply by hopeko** [reply]
> 

I see.
That means we can't get real-time data through the API, but the reports generated by Amazon.

**Reply by abuzuhri** [reply]
yes
